server/src/main/java/org/spine3/util/Environment.java

```diff
@@ -57,9 +107,76 @@ public boolean isAppEngine() {
      * Returns the current Google AppEngine version
      * or {@code null} if the program is running not on the AppEngine.
      */
-    @Nullable
-    public String getAppEngineVersion() {
-        return appEngineRuntimeVersion;
+    public Optional<String> appEngineVersion() {
+        return Optional.fromNullable(appEngineRuntimeVersion);
+    }
+
+    /**
+     * Verifies if the code currently runs under a unit testing framework.
+     *
+     * <p>The method returns {@code true} if the following packages are discovered
+     * in the stacktrace:
+     * <ul>
+     *     <li>{@code org.junit}
+     *     <li>{@code org.testng}
+     * </ul>
+     *
+     * @return {@code true} if the code runs under a testing framework, {@code false} otherwise
+     */
+    @SuppressWarnings("DynamicRegexReplaceableByCompiledPattern") // OK as we cache the result
+    public boolean isTests() {
+        // If we cached the value before, return it.
+        if (tests != null) {
+            return tests;
+        }
+
+        // Check the environment variable. We may run under unknown testing framework or
+        // tests may require production-like mode, which they simulate by setting the property to `false`.
+        String testProp = System.getProperty(ENV_KEY_TESTS);
+        if (testProp != null) {
+            testProp = testProp.replaceAll("\"' ", "");
+            this.tests = (VAL_TRUE.equalsIgnoreCase(testProp)
+                                                || "1".equals(testProp));
+            return this.tests;
+        }
+
+        // Check stacktrace for known frameworks.
+        final String stacktrace = Throwables.getStackTraceAsString(new RuntimeException(""));
+        if (stacktrace.contains("org.junit")
+                || stacktrace.contains("org.testng")) {
+            this.tests = true;
+            return true;
+        }
+
+        this.tests = false;
+        return false;
+    }
+
+    /**
+     * Turns the environment into tests mode.
```

Comment:
Perhaps, `Switches ...`?
Or maybe `Turns on the test mode for the environment.`?